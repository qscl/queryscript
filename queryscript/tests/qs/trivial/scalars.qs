let foo = 1;
SELECT foo;

let bar = [1,2,3];
select bar;

let baz = ['a', 'b', 'c'];
select baz;

select [['a', 'b'], ['c', 'd']];
select [['a', 'b'], ['c', 'd'], ];


[1,2,3];
['a', 'b', 'c'];
[['a', 'b'], ['c', 'd']];
[['a', 'b'], ['c', 'd'], ];
