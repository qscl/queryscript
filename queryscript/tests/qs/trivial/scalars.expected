{
    "compile_errors": [],
    "decls": {
        "let bar": [Int64],
        "let baz": [Utf8],
        "let foo": Int64,
    },
    "queries": [
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "foo",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| foo |\n|-----|\n| 1   |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "bar",
                                type_: List(
                                    Atom(
                                        Int64,
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| bar       |\n|-----------|\n| [1, 2, 3] |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "baz",
                                type_: List(
                                    Atom(
                                        Utf8,
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| baz       |\n|-----------|\n| [a, b, c] |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "[['a', 'b'], ['c', 'd']]",
                                type_: List(
                                    List(
                                        Atom(
                                            Utf8,
                                        ),
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| [['a', 'b'], ['c', 'd']] |\n|--------------------------|\n| [[a, b], [c, d]]         |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "[['a', 'b'], ['c', 'd']]",
                                type_: List(
                                    List(
                                        Atom(
                                            Utf8,
                                        ),
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| [['a', 'b'], ['c', 'd']] |\n|--------------------------|\n| [[a, b], [c, d]]         |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Atom(
                        Int64,
                    ),
                ),
                value: "[1, 2, 3]",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Atom(
                        Utf8,
                    ),
                ),
                value: "[a, b, c]",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    List(
                        Atom(
                            Utf8,
                        ),
                    ),
                ),
                value: "[[a, b], [c, d]]",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    List(
                        Atom(
                            Utf8,
                        ),
                    ),
                ),
                value: "[[a, b], [c, d]]",
            },
        ),
    ],
}